src/main/java/org/apache/sling/resourceresolver/impl/mapping/AliasHandler.java

```diff
@@ -130,31 +132,30 @@ protected boolean initializeAliases() {
                     } else if (!conflictingAliases.isEmpty()) {
                         log.warn("There are {} conflicting aliases: {}", conflictingAliases.size(), conflictingAliases);
                     }
+
                     if (invalidAliases.size() >= MAX_REPORT_DEFUNCT_ALIASES) {
                         log.warn("There are {} invalid aliases; excerpt: {}", invalidAliases.size(), invalidAliases);
                     } else if (!invalidAliases.isEmpty()) {
                         log.warn("There are {} invalid aliases: {}", invalidAliases.size(), invalidAliases);
                     }
                 } catch (final Exception e) {
-
+                    this.aliasMapsMap = new ConcurrentHashMap<>();
```

Comment:
That means, that in a case of an exception anywhere in this above code (most likely only in the ``loadAliases`` part) we have zero aliases? That could be considered as a breaking change.

Reply:
Hm, no. Previously we wouldn't even have used AliasHandler because of "useOptimized..." being false.

Reply:
You are correct. While before in the case of an exception ``this.aliasMapsMap`` was left with whatever was already stored there, ``mapEntries.isOptimizedAliasResolutionEnabled()``was returning false in this case and the ``aliasMapsMap`` was never consulted.

For that this is just a resource optimizations, and you could achieve the same with just a clearing of this map.
```suggestion
                    // Get rid of whatever is already stored here, it will never get used again
                    this.aliasMapsMap.clear();
```

Reply:
Actually, I'll make that an unmodifiable map for now (no additional review needed :-).